File: math/lucas_series/java/LucasSeries.java
import java.util.Scanner;

/**
 *
 * @author R. L. Ratcliffe
 * 
 * Iterative version of Lucas Sequence
 * 
 * Either enter a number as a command line argument
 * or enter a number when prompted to display n Lucas numbers.
 * 
 */
public class LucasSeries {

    public long iterativeLucas(long n) {

        long num = 2;
        long prevTerm = 1;
        long prevPrevTerm;

        if (n == 0) {
            return num;
        } else {
            for (long i = 1; i <= n; i++) {
                prevPrevTerm = num + prevTerm;
                num = prevTerm;
                prevTerm = prevPrevTerm;
            }
        }
        return num;

    }

    public static void main(String[] args) {
        LucasSeries ls = new LucasSeries();
        int n = 0;

        Scanner input = new Scanner(System.in);

        // If there is one argument, try to parse input as n
        // Otherwise, ask user to enter n
        // Amount of numbers starts at 0
        try {
            if (args.length == 1) {
                n = Integer.parseInt(args[0]);
            } else {
                System.out.println("Please enter how many numbers of the"
                                    + " Lucas series to be printed.");
                n = Integer.parseInt(input.next());
                System.out.println("");
            }
        } catch (NumberFormatException e) {
            System.err.println("Invalid amount for n.");
            System.exit(1);
        }

        // Print out sequence
        System.out.println("Here are " + n + " Lucas numbers:");
        for (long i = 0; i <= n; i++) {
            System.out.println(ls.iterativeLucas(i));
        }
    }

}

File: math/lucas_series/nodejs/LucasSeries.js
/*
    * Iterative version of Lucas Sequence

    * Lucas Numbers Formula
    L(n) = 2 (if n = 1) | 1 (if n = 2) | L(n-1) + L(n-2) (if n > 2)
*/


function LucasSeries(n) {
    // Declaring first term as 2 & second term as 1.
    var lucas_series_list = [], a1 = 2, a2 = 1;
    if (n >= 1) {
        lucas_series_list.push(a1);
    }
    if (n >= 2) {
        lucas_series_list.push(a2);
    }
    if (n > 2) {
        for (var i = 0; i < n - 2; i++) {
            var an = a1 + a2;
            a1 = a2;
            a2 = an;
            lucas_series_list.push(an);
        }
    }

    return lucas_series_list;
}

module.exports = LucasSeries;